pkg/smi/traffic_target.go

```diff
@@ -0,0 +1,168 @@
+}
+
+// Suggestion implements common.Runnable
+func (check TrafficTargetCheck) Suggestion() string {
+	return fmt.Sprintf("Check that source and desintation pod are referred to in a TrafficTarget. To get relevant TrafficTargets, use: \"kubectl get traffictarget -n %s -o yaml\"", check.dstPod.Namespace)
+}
+
+// FixIt implements common.Runnable
+func (check TrafficTargetCheck) FixIt() error {
+	panic("implement me")
+}
+
+func doesTargetMatchPods(spec access.TrafficTargetSpec, srcPod *corev1.Pod, dstPod *corev1.Pod) bool {
+	// Map traffic targets to the given pods
+	if !cli.DoesTargetRefDstPod(spec, dstPod) {
+		return false
+	}
+	// The TrafficTarget destination is associated to 'dstPod', check if 'srcPod` is an allowed source to this destination
+	if cli.DoesTargetRefSrcPod(spec, srcPod) {
+		return true
+	}
+	return false
+}
+
+// Verify interface compliance
+var _ common.Runnable = (*RoutesValidityCheck)(nil)
+
+// RoutesValidityCheck implements common.Runnable
+type RoutesValidityCheck struct {
+	cfg          configurator.Configurator
+	srcPod       *corev1.Pod
+	dstPod       *corev1.Pod
+	accessClient smiAccessClient.Interface
+}
+
+// AreTrafficRoutesValid checks whether TrafficTargets matching the src and dest pods have supported routes
+func AreTrafficRoutesValid(osmConfigurator configurator.Configurator, srcPod *corev1.Pod, dstPod *corev1.Pod, smiAccessClient smiAccessClient.Interface) RoutesValidityCheck {
+	return RoutesValidityCheck{
+		cfg:          osmConfigurator,
+		srcPod:       srcPod,
+		dstPod:       dstPod,
+		accessClient: smiAccessClient,
+	}
+}
+
+const (
+	HTTPRouteGroupKind = "HTTPRouteGroup"
+	TCPRouteKind       = "TCPRoute"
+)
+
+// Description implements common.Runnable
+func (check RoutesValidityCheck) Description() string {
+	return fmt.Sprintf("Checking whether Traffic Targets with source pod %s and destination pod %s have valid routes (%s or %s)", check.srcPod.Name, check.dstPod.Name, HTTPRouteGroupKind, TCPRouteKind)
+}
+
+// Run implements common.Runnable
+func (check RoutesValidityCheck) Run() outcomes.Outcome {
+	// Check if permissive mode is enabled, in which case every meshed pod is allowed to communicate with each other
+	if check.cfg.IsPermissiveTrafficPolicyMode() {
+		return outcomes.DiagnosticOutcome{LongDiagnostics: "OSM is in permissive traffic policy modes -- all meshed pods can communicate and SMI access policies are not applicable"}
+	}
+	trafficTargets, err := check.accessClient.AccessV1alpha3().TrafficTargets(check.dstPod.Namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return outcomes.FailedOutcome{Error: err}
+	}
+	unsupportedRouteTargets := map[string]string{}
+	for _, trafficTarget := range trafficTargets.Items {
+		spec := trafficTarget.Spec
+		if !doesTargetMatchPods(spec, check.srcPod, check.dstPod) {
+			continue
+		}
+		for _, rule := range spec.Rules {
+			kind := rule.Kind
+			if !(kind == HTTPRouteGroupKind || kind == TCPRouteKind) {
```

Comment:
This statement here is true for OSM Controllers up to v0.10.
At some point this is no longer going to be true.
We could leave this here -- but at least we need to leave a comment to update it once we have on OSM Controller, which supports other stuff.

Reply:
Yeah... I was wondering if it would make more sense to use some kind of set/mapset here instead of a hardcoded if statements. The map values could be defined as constants by picking them off each release branch, but I can't see anywhere we define them specifically as "supported routes" or something in OSM core (other than the constants you linked). For now I've added a TODO but as the list grows I agree we should change this